homeassistant/components/tesla/config_flow.py

```diff
@@ -20,76 +20,93 @@
+                errors["base"] = "cannot_connect"
+            except InvalidAuth:
+                errors["base"] = "invalid_auth"
+
+            if not errors:
+                if existing_entry:
+                    self.hass.config_entries.async_update_entry(
+                        existing_entry, data=info
+                    )
+                    await self.hass.config_entries.async_reload(existing_entry.entry_id)
+                    return self.async_abort(reason="reauth_successful")
+
+                return self.async_create_entry(
+                    title=user_input[CONF_USERNAME], data=info
+                )
+
+        return self.async_show_form(
+            step_id="user",
+            data_schema=self._async_schema(),
+            errors=errors,
+            description_placeholders={},
+        )
 
-        if not user_input:
-            return self.async_show_form(
-                step_id="user",
-                data_schema=DATA_SCHEMA,
-                errors={},
-                description_placeholders={},
-            )
-
-        if user_input[CONF_USERNAME] in configured_instances(self.hass):
-            return self.async_show_form(
-                step_id="user",
-                data_schema=DATA_SCHEMA,
-                errors={CONF_USERNAME: "already_configured"},
-                description_placeholders={},
-            )
-
-        try:
-            info = await validate_input(self.hass, user_input)
-        except CannotConnect:
-            return self.async_show_form(
-                step_id="user",
-                data_schema=DATA_SCHEMA,
-                errors={"base": "cannot_connect"},
-                description_placeholders={},
-            )
-        except InvalidAuth:
-            return self.async_show_form(
-                step_id="user",
-                data_schema=DATA_SCHEMA,
-                errors={"base": "invalid_auth"},
-                description_placeholders={},
-            )
-        return self.async_create_entry(title=user_input[CONF_USERNAME], data=info)
+    async def async_step_reauth(self, data):
+        """Handle configuration by re-auth."""
+        self.username = data[CONF_USERNAME]
+        return await self.async_step_user()
 
     @staticmethod
     @callback
     def async_get_options_flow(config_entry):
         """Get the options flow for this handler."""
         return OptionsFlowHandler(config_entry)
 
+    @callback
+    def _async_schema(self):
+        """Fetch schema with defaults."""
+        return vol.Schema(
+            {
+                vol.Required(CONF_USERNAME, default=self.username): str,
+                vol.Required(CONF_PASSWORD): str,
+            }
+        )
+
+    @callback
+    def _async_entry_for_username(self, username):
+        """Find an existing entry for a username."""
+        for entry in self._async_current_entries():
+            if entry.data.get(CONF_USERNAME) == username:
+                return entry
+
```

Comment:
Return None explicitly at the end.

Reply:
https://github.com/home-assistant/core/pull/46377